woocommerce/woocommerce-gutenberg-products-block
# Update plugin zip build script with additional options.

While working on #1780, I found the plugin zip script to be a neat tool for quickly generating a zip file for usage in testing environments where it was not possible to run the build process because npm or node was not installed (or could be installed). However, it only built production builds and for the purpose of testing I was doing, having a development build of the plugin would be more useful.

So in this pull I've extracted some of what I did in #1780 for the plugin zip script to introduce two new variations of this build script:

- `npm run package-plugin:dev` will create a development build of the plugin.
- `npm run package-plugin:zip_only` will build a zip using the current environment (useful when files are already built).

The original command (along with default behaviour for the script) continues to build production builds.

**Note:** Ordinarily I'd save doing a script update like this for the cooldown period. However, why I've done it during our cycle is because this will make it easier to distribute _development_ builds of the plugin (which will allow for more debugging information) during the testing phase.

## To Test

Try all the various package-plugin script variations and ensure:

- there are no errors
- it builds the zip using the expected mode (production, development, or existing environment).
